File: app/src/main/java/fmrsabino/moviesdb/data/local/table/MovieTable.java
package fmrsabino.moviesdb.data.local.table;

import android.content.ContentValues;
import android.database.Cursor;

import org.joda.time.DateTime;

import fmrsabino.moviesdb.data.model.movie.Movie;
import fmrsabino.moviesdb.util.CursorHelper;

public abstract class MovieTable {
    public static final String TABLE_NAME = "movie";

    public static final String COLUMN_BACKDROP_PATH = "backdrop_path";
    public static final String COLUMN_POSTER_PATH = "poster_path";
    public static final String COLUMN_RELEASE_DATE = "release_date";
    public static final String COLUMN_IMDB_ID = "imdb_id";
    public static final String COLUMN_ID = "id";
    public static final String COLUMN_TITLE = "title";
    public static final String COLUMN_TAGLINE = "tagline";
    public static final String COLUMN_OVERVIEW = "overview";


    public static final String CREATE = "CREATE TABLE " + TABLE_NAME + " (" +
            COLUMN_ID + " TEXT NOT NULL PRIMARY KEY, " +
            COLUMN_BACKDROP_PATH + " TEXT, " +
            COLUMN_POSTER_PATH + " TEXT, " +
            COLUMN_RELEASE_DATE + " TEXT, " +
            COLUMN_IMDB_ID + " TEXT, " +
            COLUMN_TITLE + " TEXT, " +
            COLUMN_TAGLINE + " TEXT, " +
            COLUMN_OVERVIEW + " TEXT" +
            " ); ";

    public static ContentValues toContentValues(Movie movie) {
        ContentValues values = new ContentValues();
        values.put(COLUMN_BACKDROP_PATH, movie.backdropPath());
        values.put(COLUMN_POSTER_PATH, movie.posterPath());
        DateTime releaseDate = movie.releaseDate();
        if (releaseDate != null) {
            values.put(COLUMN_RELEASE_DATE, releaseDate.toDate().getTime());
        }
        values.put(COLUMN_IMDB_ID, movie.imdbId());
        values.put(COLUMN_ID, movie.id());
        values.put(COLUMN_TITLE, movie.title());
        values.put(COLUMN_TAGLINE, movie.tagline());
        values.put(COLUMN_OVERVIEW, movie.overview());
        return values;
    }

    public static Movie parseCursor(Cursor c) {
        int id = CursorHelper.getInt(c, COLUMN_ID);
        if (id == CursorHelper.NULL_VALUE) {
            return null;
        }

        DateTime releaseDate = null;
        long l = CursorHelper.getLong(c, COLUMN_RELEASE_DATE);
        if (l != CursorHelper.NULL_VALUE) {
            releaseDate = new DateTime(l);
        }

        return Movie.builder()
                .backdropPath(CursorHelper.getString(c, COLUMN_BACKDROP_PATH))
                .posterPath(CursorHelper.getString(c, COLUMN_POSTER_PATH))
                .releaseDate(releaseDate)
                .imdbId(CursorHelper.getString(c, COLUMN_IMDB_ID))
                .id(CursorHelper.getInt(c, COLUMN_ID))
                .title(CursorHelper.getString(c, COLUMN_TITLE))
                .tagline(CursorHelper.getString(c, COLUMN_TAGLINE))
                .overview(CursorHelper.getString(c, COLUMN_OVERVIEW))
                .build();
    }
}

File: app/src/main/java/fmrsabino/moviesdb/data/local/table/MovieTable.kt
package fmrsabino.moviesdb.data.local.table

import android.content.ContentValues
import android.database.Cursor
import fmrsabino.moviesdb.data.model.movie.Movie
import fmrsabino.moviesdb.util.getLong
import fmrsabino.moviesdb.util.getString
import org.joda.time.DateTime

object MovieTable {
    val TABLE_NAME = "movie"

    val COLUMN_BACKDROP_PATH = "backdrop_path"
    val COLUMN_POSTER_PATH = "poster_path"
    val COLUMN_RELEASE_DATE = "release_date"
    val COLUMN_IMDB_ID = "imdb_id"
    val COLUMN_ID = "id"
    val COLUMN_TITLE = "title"
    val COLUMN_TAGLINE = "tagline"
    val COLUMN_OVERVIEW = "overview"


    val CREATE = "CREATE TABLE " + TABLE_NAME + " (" +
            COLUMN_ID + " TEXT NOT NULL PRIMARY KEY, " +
            COLUMN_BACKDROP_PATH + " TEXT, " +
            COLUMN_POSTER_PATH + " TEXT, " +
            COLUMN_RELEASE_DATE + " TEXT, " +
            COLUMN_IMDB_ID + " TEXT, " +
            COLUMN_TITLE + " TEXT, " +
            COLUMN_TAGLINE + " TEXT, " +
            COLUMN_OVERVIEW + " TEXT" +
            " ); "

    fun toContentValues(movie: Movie): ContentValues {
        val values = ContentValues()
        values.put(COLUMN_ID, movie.id)
        movie.backdropPath?.let { values.put(COLUMN_BACKDROP_PATH, movie.backdropPath) }
        movie.posterPath?.let { values.put(COLUMN_POSTER_PATH, movie.posterPath) }
        movie.releaseDate?.let { values.put(COLUMN_RELEASE_DATE, movie.releaseDate.toDate().time) }
        movie.imdbId?.let { values.put(COLUMN_IMDB_ID, movie.imdbId) }
        movie.title?.let { values.put(COLUMN_TITLE, movie.title) }
        movie.tagline?.let { values.put(COLUMN_TAGLINE, movie.tagline) }
        movie.overview?.let { values.put(COLUMN_OVERVIEW, movie.overview) }
        return values
    }

    fun parseCursor(c: Cursor): Movie {
        return Movie(
                id = c.getString(COLUMN_ID)!!,
                backdropPath = c.getString(COLUMN_BACKDROP_PATH),
                posterPath = c.getString(COLUMN_POSTER_PATH),
                releaseDate = DateTime(c.getLong(COLUMN_RELEASE_DATE)),
                imdbId = c.getString(COLUMN_IMDB_ID),
                title = c.getString(COLUMN_TITLE),
                tagline = c.getString(COLUMN_TAGLINE),
                overview = c.getString(COLUMN_OVERVIEW))
    }
}
